airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialLoadRecordIterator.java

```diff
@@ -74,11 +87,19 @@ public class MySqlInitialLoadRecordIterator extends AbstractIterator<AirbyteReco
     this.chunkSize = chunkSize;
     this.pkInfo = initialLoadStateManager.getPrimaryKeyInfo(pair);
     this.isCompositeKeyLoad = isCompositeKeyLoad;
+    this.startInstant = startInstant;
+    this.cdcInitialLoadTimeout = cdcInitialLoadTimeout;
   }
 
   @CheckForNull
   @Override
   protected AirbyteRecordData computeNext() {
+    if (isCdcSync() && cdcInitialLoadTimeout.isPresent()
+        && Duration.between(startInstant, Instant.now()).compareTo(cdcInitialLoadTimeout.get()) > 0) {
+      LOGGER.info("Sync ran for too long - sending a transient error");
+      AirbyteTraceMessageUtility.emitAnalyticsTrace(cdcSnapshotForceShutdownMessage());
+      throw new TransientErrorException("Cancelling sync because initial load taking too long");
```

Comment:
I want to spend more time on this error message (as users will see it).  How about:

```suggestion
      throw new TransientErrorException("Initial load for table ${table} has taken longer than ${initialLoadTimeoutInHours}. Canceling sync so that CDC replication can catch-up on subsequent attempt, and then initial snapshotting will resume.");
```